azure-kusto-ingest/azure/kusto/ingest/_resource_manager.py

```diff
@@ -165,3 +173,41 @@ def retrieve_service_type(self):
 
     def set_proxy(self, proxy_url: str):
         self._kusto_client.set_proxy(proxy_url)
+
+    def create_resources_by_storage_account(self, resources):
+        resources_by_storage_account = dict()
+        for resource in resources:
+            if resource.storage_account_name not in resources_by_storage_account:
+                resources_by_storage_account[resource.storage_account_name] = list()
+            resources_by_storage_account[resource.storage_account_name].append(resource)
+
+        return resources_by_storage_account
+
+    def select_with_round_robin(self, original_dict):
+        dictionary_shuffled = self.shuffle_dictionary_keys_and_values(original_dict)
+        result = []
+        while True:
+            if all(not lst for lst in dictionary_shuffled.values()):
+                break
+
+            for key, lst in dictionary_shuffled.items():
+                if lst:
+                    result.append(lst.pop(0))
+        return result
+
+    def shuffle_dictionary_keys_and_values(self, dictionary):
+        import random
+
+        keys = list(dictionary.keys())
+        random.shuffle(keys)
+
+        shuffled_dict = {}
```

Comment:
Dictionary order isn't guranteed to be insertion order.
In the python we use it happens to, but we shouldn't count on it.
Use `collections.OrderedDict`

Reply:
Didn't understand, why is it relevant?